### PR TITLE
refactor(messages): add GetSectionResponse::NotAllowed variant

### PR DESCRIPTION
**Pull request by lionel-faber**

### CODE REVIEW

File: src/section_info/mod.rs [hunk 1]
```diff
@@ -70,6 +73,9 @@ pub enum GetSectionResponse {
     /// Successful response to `GetSectionQuery`. Contains information about the requested
     /// section.
     Success(SectionInfo),
+    /// The node is not allowed to join the network
+    /// since it it not externally reachable
+    NotAllowed,
```
Comment (dirvine) [inline src/section_info/mod.rs]: Should this be `NotReachable` ?

**Review by Yoga07 — DISMISSED** [review]

File: src/section_info/mod.rs (hunk 1 above)
Comment (lionel-faber) [reply src/section_info/mod.rs]: `SectionInfoResponse::NotReachable` didn't seem very coherent to the request that's being sent.
In the future, we could expand `NotAllowed(Reason::NotReachable)` if this is being used for multiple reasons. But for now this should be okay I think?

Comment (dirvine) [reply src/section_info/mod.rs]: I don't think it is clear to readers why we are denying access though. I think specific here is maybe best?

Comment (dirvine) [reply src/section_info/mod.rs]: Or `SectionInfoResponse::NodeNotReachable`

Comment (lionel-faber) [reply src/section_info/mod.rs]: Okay. I'll make that update then :+1:

**Comment by lionel-faber** [comment]
As discussed on slack we will be taking a different path towards this. So closing for now :+1:

**Comment by lionel-faber** [comment]
Re-opened this PR so it can be used as a quickfix for testnet v6.

**Review by bochaco — APPROVED** [review]

**Pull request merged**